bokeh/charts/builder.py

```diff
@@ -688,7 +690,7 @@ def _get_range(self, dim, start, end):
 
             setattr(self, dim + 'scale', 'categorical')
             return FactorRange(factors=factors.tolist())
-        elif 'datetime' in dtype:
```

Comment:
Per http://docs.scipy.org/doc/numpy/reference/arrays.dtypes.html, this is a better check for datetime dtypes (handles array protocol types such as `dtype='<M8[ns]'`)